autometrics/src/integrations/tracing.rs

```diff
@@ -0,0 +1,95 @@
+//! Tracing integration for autometrics
+//!
+//! This module enables autometrics to use the `trace_id` field from the current span as an exemplar.
+//! Exemplars are a newer Prometheus / OpenMetrics / OpenTelemetry feature that allows you to associate
+//! specific traces with a given metric. This enables you to dig into the specifics that produced
+//! a certain metric by looking at a detailed example.
+//!
+//! # Example
+//! ```rust
+//! use autometrics::{autometrics, integrations::tracing::AutometricsLayer};
+//! use tracing::{instrument, trace};
+//! use tracing_subscriber::prelude::*;
+//! use uuid::Uuid;
+//!
+//! #[autometrics]
+//! #[instrument(fields(trace_id = %Uuid::new_v4())]
+//! fn my_function() {
+//!     trace!("Hello world!");
+//! }
+//!
+//! fn main() {
+//!     tracing_subscriber::fmt::fmt()
+//!         .finish()
+//!         .with(AutometricsLayer::default())
+//!         .init();
+//! }
+//! ```
+
+use tracing::field::{Field, Visit};
+use tracing::{span::Attributes, Id, Subscriber};
+use tracing_subscriber::layer::{Context, Layer};
+use tracing_subscriber::registry::{LookupSpan, Registry};
+
+#[derive(Clone, Hash, PartialEq, Eq, Debug, Default)]
+#[cfg_attr(
+    feature = "prometheus-client",
+    derive(prometheus_client::encoding::EncodeLabelSet)
+)]
+pub(crate) struct TraceLabel {
+    trace_id: String,
+}
+
+impl Visit for TraceLabel {
+    fn record_debug(&mut self, field: &Field, value: &dyn std::fmt::Debug) {
+        if field.name() == "trace_id" {
```

Comment:
It might be worth to make the `"trace_id"` bit a field of the `TraceLabel` struct (with a manual `impl Default for TraceLabel` to get "trace_id" by default) just so it's easier down the line to change the name of the field if necessary. I'm expecting `request_id` to be a popular option besides `trace_id`, and the adoption of the tracing feature might be better if somehow we could allow users to add the layer in an optional, non-default manner like 
```rust
tracing_subscriber::fmt::fmt()
    .finish()
    .with(AutometricsLayer::with_label("request_id"))
    .init();
```
for their own applications.

Reply:
Great suggestion!

Reply:
I added `from_field` and `from_fields` methods